esi_leap/common/flocx_market.py

```diff
@@ -0,0 +1,54 @@
+from esi_leap.db import api as dbapi
+from esi_leap.objects import flocx_market_client
+from esi_leap.objects import offer
+from keystoneauth1 import adapter
+from keystoneauth1 import loading as ks_loading
+from oslo_config import cfg
+import datetime
+
+CONF = cfg.CONF
+
+def get_flocx_market_client():
+    auth_plugin = ks_loading.load_auth_from_conf_options(
+        CONF, 'flocx_market')
+    sess = ks_loading.load_session_from_conf_options(CONF, 'flocx_market',
+                                                         auth=auth_plugin)
+    adpt = adapter.Adapter(
+        session=sess,
+        service_type='marketplace',
+        interface='public')
+    return flocx_market_client.FlocxMarketClient(adpt)
+
+def retrieve_from_flocx_market():
+    marketplace_client = get_flocx_market_client()
+    contract_list = marketplace_client.get_contracts()
+    return contract_list
+
+def get_contracts():
+    pass
+
+def update_contracts():
+    pass
+
+def update_flocx_market_contract(o_c_r_id):
+    marketplace_client = get_flocx_market_client()
+    res = marketplace_client.update_contract(o_c_r_id)
+
+def split(context, c):
+        # split contract with multiple offers into serval contracts
+    offers = c['offers']
+    o_c_rs = c['o_c_rs']
+    contracts = []
+    for i in range (len(offers)):
+        contract_dict = {}
+        o = offer.Offer.get(context, offers[i]).to_dict()
+        contract_dict['offer_uuid'] = str(o['uuid'])
+        contract_dict['project_id'] = str(o['project_id'])
+        contract_dict['start_date'] = c['start_time']
+        contract_dict['end_date'] = c['end_time']
+        contract_dict['status'] = c['status']
```

Comment:
the provider actually wants this to be set to 'open'; I would `from esi_leap.common import statuses` and set this to `statuses.OPEN`

Reply:
Oh hey, that's almost https://github.com/CCI-MOC/flocx-market/issues/121#issuecomment-518670095.

Reply:
Okay. I've set the new contract status to statuses.OPEN.